include/ofi_shm.h

```diff
@@ -123,7 +123,7 @@ struct smr_msg_hdr {
 			uint8_t	atomic_op;
 		};
 	};
-};
+} __attribute__ ((aligned(16)));
```

Comment:
I think this changes the shm protocol, and if so, requires bumping some version somewhere else.

Reply:
The reason I made this change was that the compiler was generating instructions that required 16-byte alignment for 128-bit atomics. Perhaps I can just work around the fundamental problem with a memcpy to a local variable that will have the correct alignment.

If I do have to change SMR_VERSION, which looks to be what you were referring to, the existing version check implies that I should implement backwards compatibility and I would prefer not to do that.

Reply:
For the shm provider, I doubt we need to maintain backwards compatibility.  (This isn't true for network providers, where I know we end up with mixed versions of the library).  The version check might be okay as just a sanity check.  @aingerson could chime in here on her preference.  Note that we can force 16-byte alignment here by adding a u64 reserved field, versus letting the compiler toss in invisible padding.

Does this header precede every shm message?  If so, this is a large per message header.

Reply:
I have no issue with not maintaining backwards compatibility but I brought this up a while ago and said I didn't think we needed to worry about with shm and SOMEONE (*cough* Sean *cough*) said we should.

Reply:
Thanks, I will bump SMR_VERSION; change the version check to require equality.